pandas/tests/io/test_html.py

```diff
@@ -61,9 +61,12 @@ def test_bs4_version_fails(monkeypatch, datapath):
 
 
 def test_invalid_flavor():
-    url = 'google.com'
-    with pytest.raises(ValueError):
-        read_html(url, 'google', flavor='not a* valid**++ flaver')
+    url = "google.com"
+    flavor = "invalid flavor"
+    msg = r"\{" + flavor + r"\} is not a valid set of flavors"
+
+    with tm.assert_raises_regex(ValueError, msg):
+        read_html(url, "google", flavor=flavor)
```

Comment:
We had a test for invalid flavors, but we were not checking the error message 😮 

Should definitely discourage use of `pytest.raises` (with no `match` argument) unless there is a good reason for our not wanting to check the error message.